src/org/jetbrains/plugins/scala/compiler/ScalaHydraCompilerConfigurationPanel.scala

```diff
@@ -0,0 +1,74 @@
+package org.jetbrains.plugins.scala.compiler
+
+import java.awt.event.{ActionEvent, FocusEvent, FocusListener}
+import javax.swing.event.DocumentEvent
+
+import com.intellij.openapi.ui.Messages
+import org.jetbrains.plugins.scala.caches.HydraArtifactsCache
+import org.jetbrains.plugins.scala.extensions
+import com.intellij.openapi.project.Project
+import com.intellij.ui.DocumentAdapter
+import org.jetbrains.plugins.scala.project.{Platform, ProjectExt, Version, Versions}
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success}
+
+/**
+  * @author Maris Alexandru
+  */
+class ScalaHydraCompilerConfigurationPanel(project: Project, settings: HydraCompilerSettings) extends HydraCompilerConfigurationPanel {
+  private val UnknownVersion = "unknown"
+
+  val documentAdapter = new DocumentAdapter {
+      override def textChanged(documentEvent: DocumentEvent): Unit =
+        if (!getUsername.isEmpty && !getPassword.isEmpty) downloadButton.setEnabled(true)
```

Comment:
Instead of `!getUsername.isEmpty` you can use `getUsername.nonEmpty` (same for `getPassword`